Kernel/Scheduler.cpp

```diff
@@ -211,9 +227,11 @@ bool Scheduler::pick_next()
     if (!thread_to_schedule)
         thread_to_schedule = Processor::current().idle_thread();
 
-#ifdef SCHEDULER_DEBUG
-    dbg() << "Scheduler[" << Processor::current().id() << "]: Switch to " << *thread_to_schedule << " @ " << String::format("%04x:%08x", thread_to_schedule->tss().cs, thread_to_schedule->tss().eip);
-#endif
+    if constexpr (debug_scheduler)
+        dbgln("Scheduler[{}]: Switch to {} @ {:04x}:{:08x}",
+            Processor::current().id(),
+            *thread_to_schedule,
+            thread_to_schedule->tss().cs, thread_to_schedule->tss().eip);
```

Comment:
This is a multi-line statement, it should be wrapped in curlies.

But, again, any reason not to use `dbgln<debug_scheduler>()`?

Reply:
`Processor::current().id()` is non trivial therefor I am wrapping it in `if constexpr`. I'll fix the braces.